replica/database.hh

```diff
@@ -566,24 +566,24 @@ private:
     struct merge_comparator;
 
     // update the sstable generation, making sure that new new sstables don't overwrite this one.
-    void update_sstables_known_generation(unsigned generation) {
+    void update_sstables_known_generation(sstables::generation_type generation) {
         if (!_sstable_generation) {
             _sstable_generation = 1;
         }
-        _sstable_generation = std::max<uint64_t>(*_sstable_generation, generation /  smp::count + 1);
+        _sstable_generation = std::max<uint64_t>(*_sstable_generation, sstables::generation_value(generation) / smp::count + 1);
     }
 
     sstables::generation_type calculate_generation_for_new_table() {
         assert(_sstable_generation);
         // FIXME: better way of ensuring we don't attempt to
         // overwrite an existing table.
-        return (*_sstable_generation)++ * smp::count + this_shard_id();
+        return sstables::generation_from_value((*_sstable_generation)++ * smp::count + this_shard_id());
     }
 
     // inverse of calculate_generation_for_new_table(), used to determine which
     // shard a sstable should be opened at.
-    static int64_t calculate_shard_from_sstable_generation(int64_t sstable_generation) {
-        return sstable_generation % smp::count;
+    static seastar::shard_id calculate_shard_from_sstable_generation(sstables::generation_type sstable_generation) {
+        return sstables::generation_value(sstable_generation) % smp::count;
     }
```

Comment:
I think we can potentially kill this function, probably only used in unit tests now. No need to do it in this series though. When we switch to UUID, we may no longer be able to rely on least significant bits providing a strong hint on the shard id. Today, we have scylla metadata to determine which shard(s) a given file belongs to.

Reply:
only used in "sstable_resharding_test", yes